src/whenever/__init__.py

```diff
@@ -2306,10 +2306,36 @@ def _unpkl_naive(*args) -> NaiveDateTime:
 class Ambiguous(Exception):
     """A datetime is unexpectedly ambiguous"""
 
+    @staticmethod
+    def for_timezone(d: _datetime, tz: _tzinfo) -> Ambiguous:
+        return Ambiguous(
+            f"{d.replace(tzinfo=None).isoformat()} is ambiguous "
+            f"in timezone {tz.key}"  # type:ignore[attr-defined]
+        )
+
+    @staticmethod
+    def for_system_timezone(d: _datetime) -> Ambiguous:
```

Comment:
staticmethods for alternative ways of constructing the exception—elegance 👌 ✨ 

(no change needed)